### DISCUSSION TITLE
useTemplateRef with vue 3.5+

### DISCUSSION THREAD

**Discussion by tomvdk** [quasarframework/quasar#17508] [General - Components / Directives / etc] [answered]
Hi there,

We are updating to vue 3.5+ and try to use the new [useTemplateRef feature](https://blog.vuejs.org/posts/vue-3-5#usetemplateref).
However in the following example, we are not able to call the _"hide()"_-method of the q-popup-proxy component.

![image](https://github.com/user-attachments/assets/356b3d00-3a1a-4c59-980e-0fcbf0f86c30)

As you can see it has props and slots defined but can't access a method
![image](https://github.com/user-attachments/assets/01c7f332-afb2-46db-b838-92953ef16c1f)

and therefore "casting" it to an any type but that doesn't feel right.
![image](https://github.com/user-attachments/assets/72669121-1537-42fd-9915-d21401c6b4b8)

Anyone else encountered this and know how to solve/deal with this?

Thanks!

**Edit:** When using the v-close-popup directive (as a workaround but even a better solution) I was able to trigger the close event now from the vue template. Question remains, is it possible to access the events on a quasar component when using _useTemplateRef_?

**Comment by TobyMosque** [comment, marked as answer]
do u tried to specify the type of the component?

something like that:
```ts
import type { QDate } from 'quasar'

const qDate = useTemplateRef<QDate>('qDateRef')
onMounted(() => {
  if (qDate.value) {
    qDate.value.setView('Years')
  }
})
```

**Reply by tomvdk** [reply]
Awesome! I missed you could specify the type! Thanks a lot for your quick answer!
